packages/dara-core/dara/core/internal/encoder_registry.py

```diff
@@ -80,6 +88,34 @@ def _tuple_key_serialize(d):
     return encoded_dict
 
 
+def _tuple_key_deserialize(d):
+    """
+    A helper function to recursively check if a dict have a key with "__tuple__" prefix and turn it back to tuple
+
+    :param d: the dict where tuple key to be deserialized
+    """
+    encoded_dict = {}
+    for key, value in d.items():
+        encoded_key = ast.literal_eval(key[9:]) if isinstance(key, str) and key.startswith('__tuple__') else key
```

Comment:
This is fairly risky as we're evaling data sent from the browser directly. 
According to [the docs](https://docs.python.org/3/library/ast.html#ast.literal_eval):
> This function had been documented as “safe” in the past without defining what that meant. That was misleading. This is specifically designed not to execute Python code, unlike the more general [eval()](https://docs.python.org/3/library/functions.html#eval). There is no namespace, no name lookups, or ability to call out. But it is not free from attack: A relatively small input can lead to memory exhaustion or to C stack exhaustion, crashing the process. There is also the possibility for excessive CPU consumption denial of service on some inputs. Calling it on untrusted data is thus not recommended.

Can we think of a different solution without using `eval`?